sql/ddl_internal.sql

```diff
@@ -119,10 +119,18 @@ BEGIN
 
     IF num_slices IS NULL THEN
         -- Open dimension
-        IF column_type NOT IN ('BIGINT', 'INTEGER', 'SMALLINT', 'TIMESTAMP', 'TIMESTAMPTZ') THEN
+        IF column_type NOT IN ('BIGINT', 'INTEGER', 'SMALLINT', 'DATE', 'TIMESTAMP', 'TIMESTAMPTZ') THEN
             RAISE EXCEPTION 'illegal type for column "%": %', column_name, column_type
             USING ERRCODE = 'IO102';
         END IF;
+        IF column_type = 'DATE'::regtype AND interval_length < _timescaledb_internal.interval_to_usec('1 day') THEN
```

Comment:
I'm mixed.  We already require type-specific intervals when you given an integer value, seems somewhat easier for users that:

`create_hypertable("foo", "date", ​chunk_time_interval => 7);`

As opposed to:

`create_hypertable("foo", "date", ​chunk_time_interval => 604800000000);`

Especially given that we *only* will accept answers in multiples of 86400000000.  Just bringing this up.  If others disagree, happy to approve.